### ISSUE TITLE
[Blazor WebAssembly Project Template] Change web app manifest extension to “webmanifest”

### ISSUE THREAD

**Issue by christianliebel** [dotnet/aspnetcore#43755]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

The official file extension for the Web Application Manifest is `webmanifest`. This helps servers to deliver the file with the official media type (`application/manifest+json`). I propose changing the manifest's file extension in the Blazor WebAssembly project templates from "json" to "webmanifest".

Sources:
- [Web Application Manifest spec](https://www.w3.org/TR/appmanifest/#h-note-0)
- [IANA media type registration](https://www.iana.org/assignments/media-types/application/manifest+json)

**Comment by christianliebel** [comment]
> I imagine you might be referring to the manifest used by PWAs?

@javiercn Correct, I'm referring to the Web Application Manifest used by PWAs. My PR #43756 would change the file extension of the manifest created by the project template to "webmanifest", its [official extension](https://www.iana.org/assignments/media-types/application/manifest+json).

> That would need to be handled at the static files level. @Tratcher do you have any thoughts? Would we be open to a PR that updates our mappings to support this? Do browsers follow the spec appropriately? What concrete, practical benefit does this change bring?

Yes, the static files mapping could be adjusted to include the official media type `application/manifest+json` for files with the "webmanifest" extension. For browsers, the actual file extension and media type shouldn't even matter (https://github.com/w3c/manifest/issues/689#issuecomment-450510898). So there's also no actual benefit, but it's the official file extension and media type registered at IANA.

**Comment by christianliebel** [comment]
@javiercn @Tratcher The media type is already included in the media type map: https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/StaticFiles/src/FileExtensionContentTypeProvider.cs#L347

**Comment by javiercn** [comment]
@christianliebel thanks for the clarification.

I am fine if the change is benign. Which seems to be.

**Comment by javiercn** [comment]
@christianliebel we talked about this and we will take your PR for .NET 8.0 once it passes on the CI
